src/mlpack/methods/preprocess/preprocess_imputer_main.cpp

```diff
@@ -211,11 +209,8 @@ void BINDING_FUNCTION(util::Params& params, util::Timers& timers)
       }
     }
     timers.Stop("imputation");
-
-    if (!outputFile.empty())
-    {
-      Log::Info << "Saving results to '" << outputFile << "'." << endl;
-      Save(outputFile, input, false);
-    }
+    
+    if (params.Has("output"))
+        params.Get<arma::mat>("output") = std::move(output);
```

Comment:
Just to port the comment from #3258, I think that this is incorrect---the imputer is actually working on the matrix `input`, and `output` is never set or used.  So I suspect this should be `std::move(input)`.